hypertrace-core-graphql-export-span-utils/src/main/java/org/hypertrace/core/graphql/utils/export/span/ExportSpan.java

```diff
@@ -0,0 +1,129 @@
+package org.hypertrace.core.graphql.utils.export.span;
+
+import com.google.common.io.BaseEncoding;
+import com.google.protobuf.ByteString;
+import io.opentelemetry.proto.common.v1.AnyValue;
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.resource.v1.Resource;
+import io.opentelemetry.proto.trace.v1.InstrumentationLibrarySpans;
+import io.opentelemetry.proto.trace.v1.ResourceSpans;
+import io.opentelemetry.proto.trace.v1.Span;
+import io.opentelemetry.proto.trace.v1.Span.SpanKind;
+import io.opentelemetry.proto.trace.v1.Status;
+import io.opentelemetry.proto.trace.v1.Status.StatusCode;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.Accessors;
+
+@lombok.Value
+@Accessors(fluent = true)
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public class ExportSpan {
+
+  private final ResourceSpans resourceSpans;
+
+  public static class Builder {
+
+    private static final String SERVICE_NAME_KEY = "service.name";
+    private static final String SPAN_KIND_PREFIX = "SPAN_KIND";
+    private static final String SPANK_KIND_JOINER = "_";
+
+    private final ResourceSpans.Builder resourceSpansBuilder;
+    private final Resource.Builder resourceBuilder;
+    private final InstrumentationLibrarySpans.Builder instrumentationLibrarySpansBuilder;
+    private final Span.Builder spanBuilder;
+
+    public Builder() {
+      this.resourceSpansBuilder = ResourceSpans.newBuilder();
+      this.resourceBuilder = Resource.newBuilder();
+      this.instrumentationLibrarySpansBuilder = InstrumentationLibrarySpans.newBuilder();
+      this.spanBuilder = Span.newBuilder();
+    }
+
+    public Builder setResourceServiceName(String serviceName) {
+      KeyValue keyValue =
+          KeyValue.newBuilder()
+              .setKey(SERVICE_NAME_KEY)
+              .setValue(AnyValue.newBuilder().setStringValue(serviceName).build())
+              .build();
+      this.resourceBuilder.addAttributes(keyValue);
+      return this;
+    }
+
+    public Builder setBytesFields(String spanId, String traceId, String parentSpanId) {
+      byte[] spanIdBytes = BaseEncoding.base64().decode(spanId);
```

Comment:
Is this the opposite of how we got to a string? I thought they were hex

Reply:
yes, `JsonFormat.printer` uses base64, and we are using hex.